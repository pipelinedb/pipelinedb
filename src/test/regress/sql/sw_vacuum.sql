CREATE CONTINUOUS VIEW cqvacuum AS SELECT key::text, COUNT(*) FROM cqvacuum_stream WHERE arrival_timestamp > clock_timestamp() - interval '3 second' GROUP BY key;

ACTIVATE cqvacuum;

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');

SELECT pg_sleep(1);

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');

DEACTIVATE cqvacuum;

SELECT * FROM cqvacuum ORDER BY key;

-- Just verify that the mrel has more rows, as we can't gaurantee which time bucket the
-- rows will fall in which makes it tricky to compare this result to a predetermined result
SELECT (SELECT COUNT(*) FROM cqvacuum) < (SELECT COUNT(*) FROM cqvacuum_mrel0);
SELECT DISTINCT key FROM cqvacuum_mrel0 ORDER BY key;

SELECT pg_sleep(3);

SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;

ACTIVATE cqvacuum;

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');

DEACTIVATE cqvacuum;

SELECT * FROM cqvacuum ORDER BY key;
SELECT (SELECT COUNT(*) FROM cqvacuum) < (SELECT COUNT(*) FROM cqvacuum_mrel0);
SELECT DISTINCT key FROM cqvacuum_mrel0 ORDER BY key;

VACUUM cqvacuum_mrel0;
SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;

SELECT pg_sleep(3);

SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;

VACUUM FULL cqvacuum_mrel0;
SELECT * FROM cqvacuum ORDER BY key;
SELECT key, count FROM cqvacuum_mrel0 ORDER BY key, count;

DROP CONTINUOUS VIEW cqvacuum;
