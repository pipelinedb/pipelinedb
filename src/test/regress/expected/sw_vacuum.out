CREATE CONTINUOUS VIEW cqvacuum AS SELECT key::text, COUNT(*) FROM cqvacuum_stream WHERE arrival_timestamp > clock_timestamp() - interval '3 second' GROUP BY key;
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
SELECT pg_sleep(1);
 pg_sleep 
----------
 
(1 row)

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     4
 b   |     4
 c   |     4
(3 rows)

-- Just verify that the mrel has more rows, as we can't gaurantee which time bucket the
-- rows will fall in which makes it tricky to compare this result to a predetermined result
SELECT (SELECT COUNT(*) FROM cqvacuum) < (SELECT COUNT(*) FROM cqvacuum_mrel0);
 ?column? 
----------
 t
(1 row)

SELECT DISTINCT key FROM cqvacuum_mrel0 ORDER BY key;
 key 
-----
 a
 b
 c
(3 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, SUM(count) FROM cqvacuum_mrel0 GROUP BY key ORDER BY key;
 key | sum 
-----+-----
 a   |   4
 b   |   4
 c   |   4
(3 rows)

INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
INSERT INTO cqvacuum_stream (key) VALUES ('a'), ('b'), ('c');
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

SELECT (SELECT COUNT(*) FROM cqvacuum) < (SELECT COUNT(*) FROM cqvacuum_mrel0);
 ?column? 
----------
 t
(1 row)

SELECT DISTINCT key FROM cqvacuum_mrel0 ORDER BY key;
 key 
-----
 a
 b
 c
(3 rows)

VACUUM cqvacuum;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
 a   |     2
 b   |     2
 c   |     2
(3 rows)

SELECT key, SUM(count) FROM cqvacuum_mrel0 GROUP BY key ORDER BY key;
 key | sum 
-----+-----
 a   |   2
 b   |   2
 c   |   2
(3 rows)

SELECT pg_sleep(3);
 pg_sleep 
----------
 
(1 row)

SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, SUM(count) FROM cqvacuum_mrel0 GROUP BY key ORDER BY key;
 key | sum 
-----+-----
 a   |   2
 b   |   2
 c   |   2
(3 rows)

VACUUM FULL cqvacuum;
SELECT * FROM cqvacuum ORDER BY key;
 key | count 
-----+-------
(0 rows)

SELECT key, SUM(count) FROM cqvacuum_mrel0 GROUP BY key ORDER BY key;
 key | sum 
-----+-----
(0 rows)

DROP CONTINUOUS VIEW cqvacuum;
